aries-site/src/layouts/content/SubsectionHeader.js

```diff
@@ -1,16 +1,46 @@
-import React, { useState } from 'react';
+import React, { useState, useContext, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { Box, Button, Heading } from 'grommet';
 import { Link as LinkIcon } from 'grommet-icons';
 
+import { ThemeContext } from 'styled-components';
+import { useIntersection } from '@hpe/react-hooks';
 import { HighlightPhrase } from '../../components';
 import { nameToSlug } from '../../utils';
 
+import { ActiveHeadingContext } from './ActiveHeadingContext';
+
 export const SubsectionHeader = ({ children, level }) => {
   const [over, setOver] = useState(false);
   const id = nameToSlug(children);
+  const { setActiveHeading } = useContext(ActiveHeadingContext);
+
+  const theme = useContext(ThemeContext);
+  let { large } = theme.global.edgeSize;
+  large = parseInt(large.replace('px', ''), 10); // 48
+
+  // top and bottom margin values to calculate intersection window
+  const topMargin = large;
+  const bottomMargin = `-${2 * large}`;
+
+  const options = {
+    rootMargin: `${topMargin}% 0% ${bottomMargin}% 0%`,
+    // threshold: 1.0,
+  };
+
+  const [headingToWatch, entry] = useIntersection({ options });
+  const headingIsVisible = entry.isIntersecting;
+
+  // update activeHeading, passed up to Table of Contents
+  useEffect(() => {
```

Comment:
Shouldn't this have dependencies so it's only calling setActiveHeading if id or headingIsVisible change?

Reply:
I initially had the wrong dependencies which caused the Table of Contents scroll bar not to activate, but with id and headingIsVisible it works!